File: Selenium/Activities/Activity22.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

# Start the Driver
with webdriver.Firefox() as driver:
    # Declare the wait variable
    wait = WebDriverWait(driver, timeout=10)
    # Navigate to the URL
    driver.get("https://training-support.net/webelements/popups")
    # Print the title of the page
    print("Page title is: ", driver.title)
    # Find the launcher button and click it
    driver.find_element(By.ID, "launcher").click()

    # Wait for the modal to appear
    wait.until(EC.element_to_be_clickable((By.ID, "username")))

    # Find the input fields
    username = driver.find_element(By.ID, "username")
    password = driver.find_element(By.ID, "password")
    # Enter the credentials
    username.send_keys("admin")
    password.send_keys("password")
    # Click the submit button
    driver.find_element(By.XPATH, "//button[text()='Submit']").click()

    # Print the success message
    message = driver.find_element(By.CSS_SELECTOR, "h2.text-center").text
    print("Login message: " + message)

File: Selenium/Activities/Activity22.java
package activities;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity22 {
    public static void main(String[] args) {
        // Create a new instance of the Chrome driver
        WebDriver driver = new ChromeDriver();
        // Create the Wait object
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        // Open the page
        driver.get("https://training-support.net/webelements/popups");
        // Print the title of the page
        System.out.println("Page title: " + driver.getTitle());

        // Find the launcher button and click it
        driver.findElement(By.id("launcher")).click();

        // Wait for the modal to appear
        wait.until(ExpectedConditions.elementToBeClickable(By.id("username")));

        // Find the input fields
        WebElement username = driver.findElement(By.id("username"));
        WebElement password = driver.findElement(By.id("password"));
        // Enter the credentials
        username.sendKeys("admin");
        password.sendKeys("password");
        // Click the submit button
        driver.findElement(By.xpath("//button[text()='Submit']")).click();

        // Print the success message
        String message = driver.findElement(By.cssSelector("h2.text-center")).getText();
        System.out.println("Login message: " + message);

        // Close the browser
        driver.quit();
    }
}
